packages/block-editor/src/components/link-control/index.js

```diff
@@ -146,22 +152,64 @@ function LinkControl( {
 								onChange( { ...value, ...suggestion } );
 							} }
 							isSelected={ index === selectedSuggestion }
-							isURL={ manualLinkEntryTypes.includes( suggestion.type.toLowerCase() ) }
+							isURL={ directLinkEntryTypes.includes( suggestion.type.toLowerCase() ) }
 							searchTerm={ inputValue }
 						/>
 					) ) }
+
+					{ showCreatePages && createEmptyPage && ! isInitialSuggestions && ! isSingleDirectEntryResult && (
+						<LinkControlSearchCreate
+							searchTerm={ inputValue }
+							onClick={ async () => {
+								setIsResolvingLink( true );
+								const newPage = await createEmptyPage( inputValue );
+								// TODO: handle error from API
+								setIsResolvingLink( false );
+								setIsEditingLink( false );
+								onChange( {
+									id: newPage.id,
+									title: newPage.title.raw, // TODO: use raw or rendered?
+									url: newPage.link,
+									type: newPage.type,
+								} );
+							} }
+						/>
+					) }
 				</div>
+
 			</div>
 		);
 	};
 
 	return (
 		<div className="block-editor-link-control">
-			{ ( ! isEditingLink ) && (
+
+			{ isResolvingLink && (
```

Comment:
I think we should try to avoid another duplication of this "selected' state DOM. I feel like we can extract this to a component or simply improve the logic to allow us to reuse of the existing instances.